### DISCUSSION TITLE
Error in Spatial grid

### DISCUSSION THREAD

**Discussion by Umaarasu** [giotto-suite/Giotto#742] [General]
Hi, 

Can anyone please tell why I am getting NA in the spatial network? Because of which I am not able to create a rank test. Thanks!

 visium <- createSpatialNetwork(gobject = visium,
                                      method = 'kNN', k = 5,
                                      maximum_distance_knn = 50,
                                      name = 'spatial_network')

  showGiottoSpatNetworks(visium)
└──Spatial unit "cell"
   └──S4 spatialNetworkObj "spatial_network"   (0 rows)
            from   to sdimx_begin sdimy_begin sdimx_end sdimy_end distance weight
         1: <NA> <NA>          NA          NA        NA        NA       NA     NA

 ranktest = binSpect(visium, bin_method = 'rank',
                     calc_hub = T, hub_min_int = 5,
                     spatial_network_name = 'spatial_network')
Error in `[.data.table`(full_spatial_network_DT, , `:=`(rank_int, 1:.N),  : 
  Supplied 2 items to be assigned to group 1 of size 0 in column 'rank_int'. The RHS length must either be 1 (single values are ok) or match the LHS length exactly. If you wish to 'recycle' the RHS please use rep() explicitly to make this intent clear to readers of your code.

**Comment by RubD** [comment]
Hi @Umaarasu do you have a reproducible example? And can you provide details about the version of Giotto etc? Thanks

**Comment by Umaarasu** [comment]
So, the Knn network is the one that does not work. The Delaunay network works. This issue is in my data and not the example data. I am not sure how I can get an reproducible example.

> session_info()
version  R version 4.2.1 (2022-06-23)
 os       Rocky Linux 8.8 (Green Obsidian)
 system   x86_64, linux-gnu
 ui       RStudio
─ Packages ──────────────────────────────────────────────────────
 package              * version   date (UTC) lib source
 Giotto               * 3.3.0     2023-05-05 [1] Github (drieslab/Giotto@22241fb)
 GiottoData           * 0.2.1     2023-05-05 [1] Github (drieslab/GiottoData@5e0c3f3)

showGiottoSpatNetworks(visium)
└──Spatial unit "cell"
   ├──S4 spatialNetworkObj "Delaunay_network"   (10360 rows)
   │                           from                      to sdimx_begin sdimy_begin sdimx_end sdimy_end distance      weight
   │     1: Ctrl-AAACAAGTATCTCCCA-1 Ctrl-ACTAGTTGCGATCGTC-1       34804      -28509     35385    -28509 581.0000 0.001721170
   │     2: Ctrl-AAACAAGTATCTCCCA-1 Ctrl-CAGCAGTCCAGACTAT-1       34804      -28509     34222    -28509 582.0000 0.001718213
   │     3: Ctrl-AAACAAGTATCTCCCA-1 Ctrl-GGCAGCAAACCTATGC-1       34804      -28509     35094    -29015 583.2118 0.001714643
   │     4: Ctrl-AAACAAGTATCTCCCA-1 Ctrl-AGAGGCTTCGGAAACC-1       34804      -28509     34513    -29015 583.7097 0.001713180
   │  
   └──S4 spatialNetworkObj "spatial_network"   (0 rows)
            from   to sdimx_begin sdimy_begin sdimx_end sdimy_end distance weight
         1: <NA> <NA>          NA          NA        NA        NA       NA     NA

**Reply by RubD** [reply]
HI @Umaarasu Can you share the data or a subset of the data? You could also subset a giotto object and share that. It's difficult to find a potential bug or error without being able to reproduce it and tracing it back.
